File: src/typescript/src/pipeline_services/examples/example06_mixed_local_remote.ts
import {
  Pipeline,
  RemoteSpec,
  http_step,
} from "../../index.js";
import {
  append_marker,
  normalize_whitespace,
  strip,
  to_lower,
} from "./text_steps.js";

async function main(): Promise<void> {
  const remoteSpec = new RemoteSpec(
    "http://127.0.0.1:8765/echo",
  );
  remoteSpec.method = "POST";
  remoteSpec.timeout_millis = 1000;
  remoteSpec.retries = 0;

  const pipeline = new Pipeline<string>(
    "example06_mixed_local_remote",
    true,
  )
    .addAction(strip)
    .addAction(normalize_whitespace)
    .addAction(
      (context) => http_step(remoteSpec, context),
      "remote_echo",
    )
    .addAction(to_lower)
    .addAction(append_marker);

  const output = await pipeline.run("  Hello   Remote  ");
  // eslint-disable-next-line no-console
  console.log("output=" + output);
}

void main();

File: src/Python/pipeline_services/examples/example06_mixed_local_remote.py
from __future__ import annotations

from pipeline_services.core.pipeline import Pipeline
from pipeline_services.examples.text_steps import append_marker, normalize_whitespace, strip, to_lower
from pipeline_services.remote.http_step import RemoteSpec


def main() -> None:
    fixture_endpoint = "http://127.0.0.1:8765/echo"

    remote_spec = RemoteSpec(fixture_endpoint)
    remote_spec.method = "POST"
    remote_spec.timeout_millis = 1000
    remote_spec.retries = 0

    pipeline = Pipeline("example06_mixed_local_remote", short_circuit_on_exception=True)
    pipeline.add_action(strip)
    pipeline.add_action(normalize_whitespace)
    pipeline.add_action_named("remote_echo", remote_spec)
    pipeline.add_action(to_lower)
    pipeline.add_action(append_marker)

    output_value = pipeline.run("  Hello   Remote  ")
    print("output=" + str(output_value))


if __name__ == "__main__":
    main()

